checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -380,10 +391,33 @@ public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror
         @Override
         public Void visitUnary(UnaryTree node, AnnotatedTypeMirror type) {
             // Dataflow refines this type if possible
-            type.addAnnotation(UNKNOWN);
+            if (node.getKind() == Kind.BITWISE_COMPLEMENT) {
+                addAnnotationForBitwiseComplement(
+                        getSearchIndexAnnotatedTypeFactory().getAnnotatedType(node.getExpression()),
+                        type);
+            } else {
+                type.addAnnotation(UNKNOWN);
+            }
             return super.visitUnary(node, type);
         }
 
+        private void addAnnotationForBitwiseComplement(
```

Comment:
Please document every method and field.  Otherwise I cannot review it, and more seriously future programmers (including you) won't understand it.